src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1433,6 +1432,34 @@ bool is_superset(
     return true;
 }
 
+// this function assumes that the inputs are valid schemas, and if they have a
+// $schema member it will have a string value
+bool check_compatible_dialects(
+  json::Value const& older, json::Value const& newer) {
+    auto get_dialect =
+      [](json::Value const& v) -> std::optional<json_schema_dialect> {
+        if (!v.IsObject()) {
+            // support true/false schemas
+            return std::nullopt;
+        }
+        auto it = v.FindMember("$schema");
+        if (it == v.MemberEnd()) {
+            return std::nullopt;
+        }
+        return from_uri(it->value.GetString());
+    };
+
+    auto older_dialect = get_dialect(older);
+    auto newer_dialect = get_dialect(newer);
+    if (!older_dialect.has_value() || !newer_dialect.has_value()) {
+        // no $schema, compatible
+        return true;
+    }
+    // basic support: require that both use the same dialect.
+    return older_dialect == newer_dialect;
+    // TODO: schemas using different dialects could be conditionally compatible
```

Comment:
Will we need a new ticket for this?

Reply:
yeah i'll add it

Reply:
https://redpandadata.atlassian.net/browse/CORE-6657